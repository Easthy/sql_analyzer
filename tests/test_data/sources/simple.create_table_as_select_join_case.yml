source.users:
  - id
  - name
  - status
source.user_history:
  - user_id
  - status