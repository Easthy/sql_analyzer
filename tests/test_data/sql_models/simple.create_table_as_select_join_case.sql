CREATE TABLE simple.simple.create_table_as_select_join_case AS
SELECT users.id AS user_id,
       users.name,
       CASE
            WHEN users.id IS NOT NULL
                 THEN users.status
            ELSE user_history.status
       END AS status
  FROM source.users

       LEFT JOIN source.user_history
       ON user_history.user_id = users.id
;