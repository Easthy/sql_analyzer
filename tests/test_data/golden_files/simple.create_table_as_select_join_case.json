{
  "nodes": [
    { "id": "col:simple.create_table_as_select_join_case.name" },
    { "id": "col:simple.create_table_as_select_join_case.user_id" },
    { "id": "col:simple.create_table_as_select_join_case.status" },
    { "id": "tbl:simple.create_table_as_select_join_case" },
    { "id": "tbl:source.users" },
    { "id": "col:source.users.id" },
    { "id": "col:source.users.name" },
    { "id": "col:source.users.status" },
    { "id": "tbl:source.user_history" },
    { "id": "col:source.user_history.user_id" },
    { "id": "col:source.user_history.status" }
  ],
  "links": [
    { "source": "tbl:simple.create_table_as_select_join_case", "target": "col:simple.create_table_as_select_join_case.name", "type": "contains_column" },
    { "source": "tbl:simple.create_table_as_select_join_case", "target": "col:simple.create_table_as_select_join_case.user_id", "type": "contains_column" },
    { "source": "tbl:simple.create_table_as_select_join_case", "target": "col:simple.create_table_as_select_join_case.status", "type": "contains_column" },

    { "source": "tbl:simple.create_table_as_select_join_case", "target": "tbl:source.users", "type": "table_dependency" },
    { "source": "tbl:simple.create_table_as_select_join_case", "target": "tbl:source.user_history", "type": "table_dependency" },
    
    { "source": "tbl:source.users", "target": "col:source.users.id", "type": "contains_column" },
    { "source": "tbl:source.users", "target": "col:source.users.name", "type": "contains_column" },
    { "source": "tbl:source.users", "target": "col:source.users.status", "type": "contains_column" },
    
    {"source": "col:simple.create_table_as_select_join_case.status", "target": "col:source.users.id", "type": "column_dependency"},

    { "source": "tbl:source.user_history", "target": "col:source.user_history.user_id", "type": "contains_column" },
    { "source": "tbl:source.user_history", "target": "col:source.user_history.status", "type": "contains_column" },

    { "source": "col:simple.create_table_as_select_join_case.user_id", "target": "col:source.users.id", "type": "column_dependency" },
    { "source": "col:simple.create_table_as_select_join_case.name", "target": "col:source.users.name", "type": "column_dependency" },
    { "source": "col:simple.create_table_as_select_join_case.status", "target": "col:source.users.status", "type": "column_dependency" },
    { "source": "col:simple.create_table_as_select_join_case.status", "target": "col:source.user_history.status", "type": "column_dependency" }
  ]
}